packages/vats/src/virtual-purse.js

```diff
@@ -46,22 +46,47 @@ import '@agoric/notifier/exported.js';
 function makeVirtualPurse(vpc, kit) {
   const { brand, issuer, mint, escrowPurse } = kit;
 
-  /** @type {(amt: Amount) => Promise<Payment>} */
-  let redeem;
-  /** @type {(pmt: Payment, optAmountShape?: Pattern) => Promise<Amount>} */
-  let retain;
-
-  if (mint) {
-    retain = (payment, optAmountShape) =>
-      E(issuer).burn(payment, optAmountShape);
-    redeem = amount => E(mint).mintPayment(amount);
-  } else {
+  const recoveryPurse = E(issuer).makeEmptyPurse();
+
+  /**
+   * Claim a payment for recovery via our recoveryPurse.
+   *
+   * @param {ERef<Payment>} payment
+   * @param {Pattern} [optAmountShape]
+   */
+  const recoverableClaim = async (payment, optAmountShape = undefined) => {
+    const pmt = await payment;
+    const amt = await E(recoveryPurse).deposit(pmt, optAmountShape);
+    return E(recoveryPurse).withdraw(amt);
+  };
+
+  /**
+   * @returns {{
+   *   retain: (pmt: Payment, optAmountShape?: Pattern) => Promise<Amount>,
+   *   redeem: (amt: Amount) => Promise<Payment>,
+   * }}
+   */
+  const makeRetainRedeem = () => {
+    if (mint) {
+      const retain = (payment, optAmountShape) =>
+        E(issuer).burn(recoverableClaim(payment), optAmountShape);
```

Comment:
If you are just going to burn, is there any reason to have a transition to a recoverable claim?

Reply:
Yeah, I was thinking about this last night.  No need for this recoverableClaim, nor for the other retain (in which we just deposit to the escrow purse).